cmd/sniproxy/sniproxy.go

```diff
@@ -87,8 +87,10 @@ func main() {
 
 	hostinfo.SetApp("sniproxy")
 
+	hi := hostinfo.New()
 	var s server
 	s.ts.Port = uint16(*wgPort)
+	s.ts.Hostname = hi.Hostname
```

Comment:
As a tsnet app, sniproxy can be running on a node which is also running tailscale on the host. Clearly the host's `tailscaled` should use the hostname of the host.

I do not believe that using the underlying host's hostname is the right thing to do for a tsnet app.

Reply:
Appreciate the feedback, you're of course right. I did not consider the distinction  between apps and tailscaled. I'll make the changes you suggested.